pkg/operator/staticpodoperator/controlloops/installer_controller.go

```diff
@@ -0,0 +1,377 @@
+		ret.TargetDeploymentID = 0
+
+		errors := []string{}
+		for _, containerStatus := range installerPod.Status.ContainerStatuses {
+			if containerStatus.State.Terminated != nil && len(containerStatus.State.Terminated.Message) > 0 {
+				errors = append(errors, fmt.Sprintf("%q: %s", containerStatus.Name, containerStatus.State.Terminated.Message))
+			}
+		}
+		if len(errors) == 0 {
+			errors = append(errors, "no detailed termination message, see `oc get -n %q pods/%q -oyaml`", installerPod.Namespace, installerPod.Name)
+		}
+		ret.Errors = errors
+	}
+
+	return ret, nil
+}
+
+// getDeploymentIDToStart returns the deploymentID we need to start or zero if none
+func (c *InstallerController) getDeploymentIDToStart(currKubeletState, prevKubeletState *KubeletState, operatorStatus *StaticPodConfigStatus) int32 {
+	if prevKubeletState == nil {
+		currentAtLatest := currKubeletState.CurrentDeploymentID == operatorStatus.LatestDeploymentID
+		failedAtLatest := currKubeletState.LastFailedDeploymentID == operatorStatus.LatestDeploymentID
+		if !currentAtLatest && !failedAtLatest {
+			return operatorStatus.LatestDeploymentID
+		}
+		return 0
+	}
+
+	prevInTransition := prevKubeletState.CurrentDeploymentID != prevKubeletState.TargetDeploymentID
+	if prevInTransition {
+		return 0
+	}
+
+	prevAhead := currKubeletState.CurrentDeploymentID > currKubeletState.CurrentDeploymentID
+	failedAtPrev := currKubeletState.LastFailedDeploymentID == prevKubeletState.CurrentDeploymentID
+	if prevAhead && !failedAtPrev {
+		return currKubeletState.CurrentDeploymentID
+	}
+
+	return 0
+}
+
+func getInstallerPodName(deploymentID int32, nodeName string) string {
+	return fmt.Sprintf("installer-%d-%s", deploymentID, nodeName)
+}
+
+// createInstallerPod creates the installer pod with the secrets required to
+func (c *InstallerController) createInstallerPod(currKubeletState *KubeletState, operatorSpec *operatorv1alpha1.OperatorSpec, deploymentID int32) (*KubeletState, error) {
+	required := resourceread.ReadPodV1OrDie([]byte(installerPod))
+	switch corev1.PullPolicy(operatorSpec.ImagePullPolicy) {
+	case corev1.PullAlways, corev1.PullIfNotPresent, corev1.PullNever:
+		required.Spec.Containers[0].ImagePullPolicy = corev1.PullPolicy(operatorSpec.ImagePullPolicy)
+	case "":
+	default:
+		return nil, fmt.Errorf("invalid imagePullPolicy specified: %v", operatorSpec.ImagePullPolicy)
+	}
+	required.Name = getInstallerPodName(deploymentID, currKubeletState.NodeName)
+	required.Spec.NodeName = currKubeletState.NodeName
+	required.Spec.Containers[0].Image = getInstallerPodImage()
+	required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args,
+		fmt.Sprintf("-v=%d", operatorSpec.Logging.Level),
+		fmt.Sprintf("--deployment-id=%d", deploymentID),
+		fmt.Sprintf("--namespace=%s", c.targetNamespace),
+		fmt.Sprintf("--pod=%s", c.configMaps[0]),
+		fmt.Sprintf("--resource-dir=%s", "/etc/kubernetes/static-pod-resources"),
```

Comment:
i think the defaults here should be consts

Reply:
> i think the defaults here should be consts

why?